tools/codesigningtool/codesigningtool.py

```diff
@@ -120,16 +123,100 @@ def _certificate_fingerprint(identity):
   fingerprint = fingerprint.replace(":", "")
   return fingerprint
 
+def _certificate_common_name(cert):
+  _, subject, _ = execute.execute_and_filter_output([
+    "openssl",
+    "x509",
+    "-noout",
+    "-inform",
+    "DER",
+    "-subject"
+  ], inputstr=cert, raise_on_failure=True)
+  subject = subject.strip().split('/')
+  cert_cn = [f for f in subject if "CN=" in f][0]
+  cert_cn = cert_cn.replace("CN=", "")
+  return cert_cn
 
 def _get_identities_from_provisioning_profile(mpf):
   """Iterates through all the identities in a provisioning profile, lazily."""
   for identity in mpf["DeveloperCertificates"]:
-    if not isinstance(identity, bytes):
+    cert = _certificate_data(identity)
+    yield _certificate_fingerprint(cert)
+
+def _certificate_data(cert):
+  if not isinstance(cert, bytes):
       # Old versions of plistlib return the deprecated plistlib.Data type
       # instead of bytes.
-      identity = identity.data
-    yield _certificate_fingerprint(identity)
+      cert = cert.data
+
+  return cert
```

Comment:
I don't think we would need this validation as long as we aren't using `plistlib` to parse the output, and so shouldn't be getting any data/bytes types.

Reply:
I just moved existing code. [This came from an old fix](https://github.com/bazelbuild/rules_apple/commit/11aa8d7fd2f99ff8841f29dfaee16b2ded9955bc) re different versions `plistlib` being used, not sure if we can just stop doing the check?

Reply:
I think so, because in parsing XML elements you should be only dealing with strings, whereas with plists objects could be stored as data or other types.

Reply:
I am now using `plistlib` so I'll keep this